src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -138,18 +138,29 @@ class ModelQueue {
   }
 
   std::vector<PredictTask> get_batch(
-      std::function<int(Deadline)> &&get_batch_size) {
+      std::function<size_t(Deadline)> &&get_batch_size) {
     std::unique_lock<std::mutex> lock(queue_mutex_);
     remove_tasks_with_elapsed_deadlines();
     queue_not_empty_condition_.wait(lock, [this]() { return !queue_.empty(); });
     remove_tasks_with_elapsed_deadlines();
     Deadline deadline = queue_.top().first;
-    int max_batch_size = get_batch_size(deadline);
+    size_t max_batch_size = get_batch_size(deadline);
     std::vector<PredictTask> batch;
-    while (batch.size() < (size_t)max_batch_size && queue_.size() > 0) {
+    while (batch.size() < max_batch_size && queue_.size() > 0) {
       batch.push_back(queue_.top().second);
       queue_.pop();
     }
+    size_t batch_differential = max_batch_size - batch.size();
+    if (batch_differential > 0) {
+      // Artificially inject queries to create
+      // a full batch
+      //
+      // Create a copy of the last task in the batch and make it artifical
+      PredictTask last_task = batch.back();
+      last_task.artificial_ = true;
+
+      std::fill_n(std::back_inserter(batch), batch_differential, last_task);
+    }
```

Comment:
We should only inject artificial padding queries if we are exploring. The idea would be that once the system has explored sufficiently and is running at steady state, no more artificial queries are being injected.